client/state/sites/selectors.js

```diff
@@ -68,36 +67,41 @@ export const getSiteBySlug = createSelector(
 );
 
 /**
- * Returns a normalized site object by its ID. Intends to replicate
- * the site object returned from the legacy `sites-list` module.
- *
+ * Returns a normalized site object by its ID or site slug.
  *
  * @param  {Object}  state  Global state tree
- * @param  {Number}  siteId Site ID
+ * @param  {Number}  siteId Site ID or site slug
  * @return {?Object}        Site object
  */
-export const getSite = createSelector(
-	( state, siteId ) => {
-		let site =
-			getRawSite( state, siteId ) ||
-			// Support for non-ID site retrieval
-			// Replaces SitesList#getSite
-			getSiteBySlug( state, siteId );
-
-		if ( ! site ) {
-			return null;
-		}
+let getSiteCache = new WeakMap();
```

Comment:
should this be above the docblock?

Reply:
we can also give it its own JSDOC block comment indicating its use and contents

Reply:
Docblock rearranged, added own docblock for `getSiteCache`